<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Junwei Han" />


<title>ICDS User Guide</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">ICDS User Guide</h1>
<h4 class="author">Junwei Han</h4>



<div id="introduce" class="section level2">
<h2>1 Introduce</h2>
<p>This vignette illustrates how to easily use the ICDS package. This
package can identifying cancer risk subpathways which can explain the
disturbed biological functions in more detail and accurately.</p>
<ul>
<li><p>This package provides the
<code>getExpp</code>,<code>getMethp</code>,<code>getCnvp</code>function
to calculate p-values or corrected p-values for each gene.</p></li>
<li><p>This package provides the
<code>coverp2zscore</code>,<code>combinep_two</code>,<code>combinep_three</code>
function to Convert p-values or corrected p-values to z-scores.</p></li>
<li><p>This package provides the <code>FindSubPath</code> function to
search for interested subpathways in each entire pathway.</p></li>
<li><p>This package provides the <code>opt_subpath</code> function to
Optimize interested subpathways.</p></li>
<li><p>This package provides the <code>Permutation</code> function to to
calculate statistical significance for these interested subpathways
.</p></li>
</ul>
</div>
<div id="example-obtain-p-values-or-corrected-p-values-for-each-gene" class="section level2">
<h2>2 Example: Obtain p-values or corrected p-values for each gene</h2>
<p>We can use function getExampleData to return example data and
environment variables, such as the data of exp_data.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(ICDS)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="co"># obtain the expression profile data</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>exp_data<span class="ot">&lt;-</span><span class="fu">GetExampleData</span>(<span class="st">&quot;exp_data&quot;</span>)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="co">#view first six rows and six colmns of data</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>exp_data[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span></code></pre></div>
<pre><code>##       TCGA.38.4632.11 TCGA.44.5645.11 TCGA.44.6144.11 TCGA.44.6145.11
## GALM      15.55599000      18.3477000      13.4976300      12.4355000
## HK3        9.65274100      12.4103200      12.3639500      17.5058100
## G6PC       0.06917292       0.1689286       0.1555786       0.1847281
## PGM1      24.33696000      69.3011200      68.6039000      48.9442400
## HKDC1      0.65847880       1.2588160       3.0050750       2.8253770
## GPI       73.56174000      58.7149700      62.7976900      58.1940700
##       TCGA.44.6146.11 TCGA.44.6147.11
## GALM       16.5563600       8.3897860
## HK3         6.8267430       8.2327620
## G6PC        0.1555417       0.1721295
## PGM1       32.4310600      43.4132300
## HKDC1       1.1371050       0.5703535
## GPI        53.6260600      41.9420700</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co">#obtain the labels of the samples of the expression profile, the label vector is a vector of 0/1s,</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="co"># 0 represents the case sample and 1 represents the control sample</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>label1<span class="ot">&lt;-</span><span class="fu">GetExampleData</span>(<span class="st">&quot;label1&quot;</span>)</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="co">#view first ten label</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>label1[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span></code></pre></div>
<pre><code>##  [1] 0 0 0 0 0 0 0 0 0 0</code></pre>
<p>we used the Student’s t-test to calculate the p-value for expression
level and methylation level of each gene in the tumor and normal
samples.label,0 represents normal samples;1 represents cancer
samples.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="co">#calculate p-values or corrected p-values for each gene</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>exp.p<span class="ot">&lt;-</span><span class="fu">getExpp</span>(exp_data,<span class="at">label =</span> label1,<span class="at">p.adjust =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>label2<span class="ot">&lt;-</span><span class="fu">GetExampleData</span>(<span class="st">&quot;label2&quot;</span>)</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>meth_data<span class="ot">&lt;-</span><span class="fu">GetExampleData</span>(<span class="st">&quot;meth_data&quot;</span>)</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>meth.p<span class="ot">&lt;-</span><span class="fu">getMethp</span>(meth_data,<span class="at">label =</span> label2,<span class="at">p.adjust =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>exp.p[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,]</span></code></pre></div>
<pre><code>##            gene                    p
## GALM       GALM  0.00382104698965552
## HK3         HK3 1.22893635962516e-14
## G6PC       G6PC    0.309765961639907
## PGM1       PGM1 6.50659302913753e-05
## HKDC1     HKDC1 1.27479647059785e-05
## GPI         GPI 1.37454189773835e-11
## FBP1       FBP1 2.09855354282047e-12
## ALDOA     ALDOA 2.54043616919423e-14
## G6PC3     G6PC3 0.000265142879987233
## ALDH7A1 ALDH7A1  0.00626937915913354</code></pre>
<p>For copy number data, we can calculate p-value through the following
way:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="co">#obtain Copy number variation data</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>cnv_data<span class="ot">&lt;-</span><span class="fu">GetExampleData</span>(<span class="st">&quot;cnv_data&quot;</span>)</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="co">#obtion amplified genes</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>amp_gene<span class="ot">&lt;-</span><span class="fu">GetExampleData</span>(<span class="st">&quot;amp_gene&quot;</span>)</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="co">#obtion deleted genes</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>del_gene<span class="ot">&lt;-</span><span class="fu">GetExampleData</span>((<span class="st">&quot;del_gene&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="co">#calculate p-values or corrected p-values for each gene</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>cnv.p<span class="ot">&lt;-</span><span class="fu">getCnvp</span>(exp_data,cnv_data,amp_gene,del_gene,<span class="at">p.adjust=</span><span class="cn">FALSE</span>,<span class="at">method=</span><span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>cnv.p[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,]</span></code></pre></div>
<pre><code>##            gene                   p
## ACSS1     ACSS1 0.00641326042511501
## ACSS2     ACSS2                   1
## ADH1A     ADH1A   0.454656092798328
## ADH1B     ADH1B   0.429492188406724
## ADH5       ADH5 0.00253178462835118
## AKR1A1   AKR1A1                   1
## ALDH1A3 ALDH1A3                   1
## ALDH1B1 ALDH1B1                   1
## ALDH2     ALDH2   0.145658828705581
## ALDH3B1 ALDH3B1                   1</code></pre>
</div>
<div id="example-combine-p-values-of-different-kinds-of-data" class="section level2">
<h2>3 Example: Combine P-values of different kinds of data</h2>
<p>With the above data, we constructed an integrative gene z-score (Z)
based three datasets.</p>
<ul>
<li><p>Firstly,the gene score calculated through combined p-values of
Fisher’s Inverse chi-square tests. This method computes a combined
statistic S from the p-values of the difference coefficient obtained
from three individual datasets .Usually,the statistic S follows a
chi-square distribution with 2k degrees,we can calculate the null
hypothesis p-value of the statistic S.</p></li>
<li><p>Secondly,we convert the p-value to z-score according to Gaussian
distribution , which is taken as the gene z-score(Z) .</p></li>
</ul>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="co">#obtain the p-values of expression profile data,methylation profile data and Copy number variation data</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>exp.p<span class="ot">&lt;-</span><span class="fu">GetExampleData</span>(<span class="st">&quot;exp.p&quot;</span>)</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>meth.p<span class="ot">&lt;-</span><span class="fu">GetExampleData</span>(<span class="st">&quot;meth.p&quot;</span>)</span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>cnv.p<span class="ot">&lt;-</span><span class="fu">GetExampleData</span>(<span class="st">&quot;cnv.p&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="co">#calculate z-scores for p-values of each kind of data</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>zexp<span class="ot">&lt;-</span><span class="fu">coverp2zscore</span>(exp.p)</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>zmeth<span class="ot">&lt;-</span><span class="fu">coverp2zscore</span>(meth.p)</span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a>zcnv<span class="ot">&lt;-</span><span class="fu">coverp2zscore</span>(cnv.p)</span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a><span class="co">#combine two kinds of p-values,then,calculate z-score for them</span></span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a>zz<span class="ot">&lt;-</span><span class="fu">combinep_two</span>(exp.p,meth.p)</span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a><span class="co">#combine three kinds of p-values,then,calculate z-score for them</span></span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a>zzz<span class="ot">&lt;-</span><span class="fu">combinep_three</span>(exp.p,meth.p,cnv.p)</span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a>zzz[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>,]</span></code></pre></div>
<pre><code>##                       padjust          z_score                exp.p
## AACS     1.49856652283489e-12 6.97785849233781 4.27591802855694e-05
## AASDHPPT 3.59090577618182e-09 5.78661689489162 1.51758714809107e-05
## ACAA1    2.26766115435344e-06 4.58520914055579  0.00254706069215554
## ACACA    7.81825182198367e-05 3.78073660442176   0.0414461349050663
## ACACB    3.17474389450551e-24 10.0863369031681  1.0796730435591e-25
## ACAD8    8.80185318920442e-08 5.22301085773967 2.21022111058924e-05
##                        meth.p                cnv.p
## AACS     3.79351155555026e-07 0.000154005933715148
## AASDHPPT    0.138052336368975 4.95190895519194e-06
## ACAA1    4.80814465758352e-06                    1
## ACACA    1.62339209249856e-05                    1
## ACACB       0.152770581756488   0.0979253199784077
## ACAD8        0.19898879195769 7.67894501521561e-05</code></pre>
</div>
<div id="example-obtain-subpathways" class="section level2">
<h2>4 Example: Obtain subpathways</h2>
<p>For a priori KEGG path, we perform a greedy algorithm to search for
interested subpathways.We provide two statistical test methods of the
subpathway, which one is whole gene-based perturbation, and the other is
the local gene perturbation in a particular pathway.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="co">#obtain z-score of each gene</span></span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>zzz<span class="ot">&lt;-</span><span class="fu">GetExampleData</span>(<span class="st">&quot;zzz&quot;</span>)</span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a>zzz[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,]</span></code></pre></div>
<pre><code>##                       padjust          z_score                exp.p
## AACS      1.4985665228349e-12 6.97785849233781 4.27591802855694e-05
## AASDHPPT 3.59090577618182e-09 5.78661689489162 1.51758714809107e-05
## ACAA1    2.26766115435345e-06 4.58520914055579  0.00254706069215554
## ACACA    7.81825182198368e-05 3.78073660442176   0.0414461349050663
## ACACB    3.17474389450549e-24 10.0863369031681  1.0796730435591e-25
## ACAD8    8.80185318920443e-08 5.22301085773967 2.21022111058924e-05
## ACADL    6.14820533862155e-54 15.4184802326598 2.63073053599382e-55
## ACAT1     1.1545332900424e-06 4.72430235586153 6.58517201812735e-06
## ACER1    2.96364608769913e-06  4.5289664035691                    1
## ACER3    1.63555761386473e-07 5.10711682637474  5.9116689567922e-07
##                        meth.p                cnv.p
## AACS     3.79351155555026e-07 0.000154005933715148
## AASDHPPT    0.138052336368975 4.95190895519194e-06
## ACAA1    4.80814465758352e-06                    1
## ACACA    1.62339209249856e-05                    1
## ACACB       0.152770581756488   0.0979253199784077
## ACAD8        0.19898879195769 7.67894501521561e-05
## ACADL     0.00265801159320894                    1
## ACAT1       0.287245454170909  0.00305344462196611
## ACER1    1.65137200042873e-08                    1
## ACER3     0.00112767915011995                    1</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="fu">require</span>(graphite)</span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>zz<span class="ot">&lt;-</span><span class="fu">GetExampleData</span>(<span class="st">&quot;zzz&quot;</span>)</span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a><span class="co">#subpathdata&lt;-FindSubPath(zz) #only show</span></span></code></pre></div>
<p>Optimize interested subpathways.If the number of genes shared by the
two pathways accounted for more than the Overlap ratio of each pathway
genes,than combine two pathways.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>subpathdata<span class="ot">&lt;-</span><span class="fu">GetExampleData</span>(<span class="st">&quot;subpathdata&quot;</span>)</span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>keysubpathways<span class="ot">&lt;-</span><span class="fu">opt_subpath</span>(subpathdata,zz,<span class="at">overlap=</span><span class="fl">0.6</span>)</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a><span class="fu">head</span>(keysubpathways)</span></code></pre></div>
<pre><code>##      SubpathwayID  pathway                       
## [1,] &quot;path00010_1&quot; &quot;Glycolysis / Gluconeogenesis&quot;
## [2,] &quot;path00010_2&quot; &quot;Glycolysis / Gluconeogenesis&quot;
## [3,] &quot;path00010_3&quot; &quot;Glycolysis / Gluconeogenesis&quot;
## [4,] &quot;path00010_4&quot; &quot;Glycolysis / Gluconeogenesis&quot;
## [5,] &quot;path00010_5&quot; &quot;Glycolysis / Gluconeogenesis&quot;
## [6,] &quot;path00010_6&quot; &quot;Glycolysis / Gluconeogenesis&quot;
##      Subgene                                                                
## [1,] &quot;G6PC/HK3/PGM1/HKDC1/GPI/FBP1/ALDOA/G6PC2/GALM/G6PC3&quot;                  
## [2,] &quot;ADH5/ADH1B/ADH1A/ALDH3B2/ALDH2/AKR1A1/ALDH7A1/ALDH3B1/ALDH1B1/ALDH1A3&quot;
## [3,] &quot;PKLR/ENO1/LDHA/PGAM1/MINPP1/PCK1/LDHAL6A/LDHAL6B&quot;                     
## [4,] &quot;ACSS1/ALDH3B2/ADH1B/ADH1A/ALDH2/DLAT/ACSS2&quot;                           
## [5,] &quot;PFKP/ALDOA/GAPDH/FBP1/GPI/ALDOC/GAPDHS&quot;                               
## [6,] &quot;PGAM4/ENO1/PGAM1/MINPP1/PCK1/PGK2&quot;                                    
##      Size SubpathwayZScore  
## [1,] &quot;10&quot; &quot;20.8492758044976&quot;
## [2,] &quot;10&quot; &quot;20.9317587858665&quot;
## [3,] &quot;8&quot;  &quot;19.0899605586423&quot;
## [4,] &quot;7&quot;  &quot;19.7078248333379&quot;
## [5,] &quot;7&quot;  &quot;18.3020024323356&quot;
## [6,] &quot;6&quot;  &quot;16.7772761794305&quot;</code></pre>
<p>the perturbation test was used to calculate statistical significance
for these interested subpathways.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a>keysubpathways<span class="ot">&lt;-</span><span class="fu">Permutation</span>(keysubpathways,zz,<span class="at">nperm1=</span><span class="dv">100</span>,<span class="at">method1=</span><span class="cn">TRUE</span>,<span class="at">nperm2=</span><span class="dv">100</span>,<span class="at">method2=</span><span class="cn">FALSE</span>)</span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a><span class="fu">head</span>(keysubpathways)</span></code></pre></div>
<pre><code>##      SubpathwayID  pathway                       
## [1,] &quot;path00010_1&quot; &quot;Glycolysis / Gluconeogenesis&quot;
## [2,] &quot;path00010_2&quot; &quot;Glycolysis / Gluconeogenesis&quot;
## [3,] &quot;path00010_3&quot; &quot;Glycolysis / Gluconeogenesis&quot;
## [4,] &quot;path00010_4&quot; &quot;Glycolysis / Gluconeogenesis&quot;
## [5,] &quot;path00010_5&quot; &quot;Glycolysis / Gluconeogenesis&quot;
## [6,] &quot;path00010_6&quot; &quot;Glycolysis / Gluconeogenesis&quot;
##      Subgene                                                                
## [1,] &quot;G6PC/HK3/PGM1/HKDC1/GPI/FBP1/ALDOA/G6PC2/GALM/G6PC3&quot;                  
## [2,] &quot;ADH5/ADH1B/ADH1A/ALDH3B2/ALDH2/AKR1A1/ALDH7A1/ALDH3B1/ALDH1B1/ALDH1A3&quot;
## [3,] &quot;PKLR/ENO1/LDHA/PGAM1/MINPP1/PCK1/LDHAL6A/LDHAL6B&quot;                     
## [4,] &quot;ACSS1/ALDH3B2/ADH1B/ADH1A/ALDH2/DLAT/ACSS2&quot;                           
## [5,] &quot;PFKP/ALDOA/GAPDH/FBP1/GPI/ALDOC/GAPDHS&quot;                               
## [6,] &quot;PGAM4/ENO1/PGAM1/MINPP1/PCK1/PGK2&quot;                                    
##      Size SubpathwayZScore   p1     fdr1               
## [1,] &quot;10&quot; &quot;20.8492758044976&quot; &quot;0.12&quot; &quot;0.534260869565217&quot;
## [2,] &quot;10&quot; &quot;20.9317587858665&quot; &quot;0.08&quot; &quot;0.465454545454545&quot;
## [3,] &quot;8&quot;  &quot;19.0899605586423&quot; &quot;0.06&quot; &quot;0.420821917808219&quot;
## [4,] &quot;7&quot;  &quot;19.7078248333379&quot; &quot;0.01&quot; &quot;0.155151515151515&quot;
## [5,] &quot;7&quot;  &quot;18.3020024323356&quot; &quot;0.09&quot; &quot;0.495483870967742&quot;
## [6,] &quot;6&quot;  &quot;16.7772761794305&quot; &quot;0.04&quot; &quot;0.372363636363636&quot;</code></pre>
</div>
<div id="example-plot-a-network-graph-when-user-input-a-list-of-gene" class="section level2">
<h2>4 Example: plot a network graph when user input a list of gene</h2>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="fu">require</span>(graphite)</span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a><span class="fu">require</span>(org.Hs.eg.db)</span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a>subpID<span class="ot">&lt;-</span><span class="fu">unlist</span>(<span class="fu">strsplit</span>(<span class="st">&quot;ACSS1/ALDH3B2/ADH1B/ADH1A/ALDH2/DLAT/ACSS2&quot;</span>,<span class="st">&quot;/&quot;</span>))</span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a>pathway.name<span class="ot">=</span><span class="st">&quot;Glycolysis / Gluconeogenesis&quot;</span></span>
<span id="cb20-5"><a href="#cb20-5" tabindex="-1"></a>zzz<span class="ot">&lt;-</span> <span class="fu">GetExampleData</span>(<span class="st">&quot;zzz&quot;</span>)</span>
<span id="cb20-6"><a href="#cb20-6" tabindex="-1"></a><span class="fu">PlotSubpathway</span>(<span class="at">subpID=</span>subpID,<span class="at">pathway.name=</span>pathway.name,<span class="at">zz=</span>zzz)</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAABoVBMVEUAAAAAACAAADgAAE4AAGMAAGUAAG0AAHcAAHkAAIsAAJEAAJIAAJUAAJkAAJ0AAJ8AAK0AALMABD0AG48AJpsAOp8AOrMAOscAOs4ARJ8AWNoAZtoIAIsXAIsXZu0hAIsmAIsmAJEmX6Q4CCY6AIs6AJk6AJ86ALM6F4s6KJ86Osc6Zr46Zto6e706kO1EAItEealJAItJOotYOotYtv9fJotfealfkaliSGViXoNicqBjAHZkCg1mAItmAJ9mAK1mALNmF4tmOotmZotmnMdmncdmtv95RIt5qal7IIuJiqCKF3+KF4uKHYiKOXKKSoaLPkiMCw2QOouQOp+QOrOQZseQkNqQnbOQtv+Q2/+RX5GRqamdOoupeZWpkZupqZ+pqaSpqamuoKCuvr6wSEiwZWWxKiq2Zou2Zp+2/9q2/+22///SWo/Sa5PTOYTVGHzWCHnbkJ/bkMfb/+3b///0oKD0vr71Xoj1cYf1g4P1g4T2ZWX2ZWz3M2v3SEj3SF/4Kir5DQ3/tq3/trP/tsf/vKn/28f/2+3//9r//+3////j672pAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAJt0lEQVR4nO2dCXvcRgGG5cRgVFJCIWBMOIoxlHMTjhIuQ7jWoaUFTCndhLQM1wJuoSpOwHUIDQqLV7+auSSNjvWXrXWMnv3ePFlrtet4583MSJr5Rg4ScipB3x/AdygIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCAEMTFF+cyMeT3VBzOUnm+/LrOInSZ5uH6av6nWdmaIKm4Uh/jZ44kA+hflQPyWxnrMVJQXKXfJzvN2JoYIJOrv9AG7BWYuXDbJ7sjvX+TJBUNmrgJw5MUDQybcxaSaby2SJBJ7urJ2j+i4ktthUUheNgY71eULSCTSyW/fBUq7GC4nDtzqumww5dQeq5ecdZGZag6Vg5USayGvTNO6+++7dBbQ0KV66JzXbSg3vWB208s1DQfL+JOjQoQZGuEtP82B6H5+4sFKR78DMzJEGzp3SN0Ifv7DwokIKCxJ4HSSluE7vcwA8dkCDZwFSVUO3sA6atqU7mN6FzJq3OoS9OnPPsszMgQbUI+SfRf1ti4IKMHwpahEjdtGdo0ILy1kVBtTjdT2uGhiyo0D23ZWjAgordMwWVKXc+LRkarKBK30NBBWr65nYMDVSQKH0tbzfHMAXVd82tGBq2IFG/u0kGKWjRsZ2CDIuPXC0YGqAgUbtZ+7wBhifoFD9tGBqcIHBgb9zQ0ASVj+vlMIOoCTNo4rc5gu+pIFMcXVg1k5EJEIU0h1D7P6a/IRvEF3VhhiRPPSyLp4IKIY5sIwovuGkOMX/hz9e/En41f2Mcbt6qm6t3Ug/L4qegPMSRlsqU+nYhzSEeTqLRf8IP56+rqTCxsR4kFUFZ6mFZ/BSUhzgKgkQxzSF0mOFLH8peN2GG197zu/JEopN6WBYvBTkhjlSQqhZCC8gEXdVhhvn3SmGGu69VwwxO6mFZvBTkhDisoOBv7/yj+qix7JsiK+CqDjPMPqtN3LxwQ0huhsFdWYPuBqUalKcelsVLQU6Iw5QveEsKeivIapBU8cqVq+LlJ7WpLfk064NqBTmph2XxUZAb4igIEmLvsWtCVpbENLFIVqdxMcxwWdT1QU7qYVl8FOSGOHT5HqY1KFbWbB90ZTx76iCWb/3vx50www3TXRXDDG7qYVk8FFQJcZxcv2H6oEKq9ckf7oSP2e4oDTNsF08kbZjBST0sfy7knyCnOB/8vin3VnomPSrmoqNSz7Ki40HZ/Jeo7gZ7GsB7QSKfHzx9AFqs5qB9YXrntKGO1ZxZzWoFFLSa4YXK4POi0eiWmpfCY0GiqmOBoBYDZh4Lqm1PtRM+bfrxVpCoryI1glpsXgpPBS3sgrsKdWR4KahcKaozPa3npjJ8FGQK7Yzbbyf5uP1W4WpjS+8fF1Zhmj1N0Y+gfHw4lsXa/PsvD9U4fTiavST17JmNafhRXVA1/KPf6q7CvGCuV6WZ6wf2hXQV5vyFAz1s1BD9CMrmYKbmulSWUI1LqHIJu3Hy3c9sHurxsc1DR5BZhXlTb76yO07+p5YWfmfirMJ8OFF1rJFlCIpeBOWTFuZ/WlYDM741+5qwGy9Fn//VxYkqaLR5y3yXHTTck06VIGHfKt/8haS8CnM6bEHppMXJrh2feSNYC820RbrKy47bT6uCojAINnSzs4Jmn6uswpzvD7qJZZMW2RBfcHT0Dz3cleheaZSO2/9lf1wRFIeffPC6M3ERmcWXhVWYxlkj9CEom7SIbVck/Rwd/TOdqVFDZWMzbv8jWdCyoI3w2w9ef4czN2jWXhZWYf6pkXsKmM/W2L/06GSTFgVBR8FtVYkUcuMjuoa8/5oQNy/d0juFndkJ3/XzkqD5firIts+ouaN8H4LySYuTXdOXSkFv/vQo0OWbvyh3zH/8XlXQX+u2c0mP20+yo9j7HpQEaePOKkw1kq9OGBqhB0HOpEW6qnTjG28+p+cFpSB5GpPMPqVb4Bd35AmRPufJT3fCJwIpaD0LcCS2x8lXYcbNrQlP+hDkTlrIc2JVxunIHMXUUV3VIlnHHp+Ygj7+ZRt8ycftxZqzClOfaB46AaGkPJJ/RjoXVJqD0cVRMYQXVRlV5CndSAva1djqAvy4Fjut0F2NrS7AC0GnFpqClhLUtR8vBPnsZ2CCuvfjg6DTS93Z6PwCKAjQvyC//QxJUC9+BiSoHz/9CwLlpiDPK1Dvgh61AvXlZyiCevPTtyBU8MpUfOf4Laj/CtSzIP8r0CAE9emnX0Gw5NX0b+d4LcgDPxSE6FPQI7Wwnv34I6guU7UXhtt6ijQNVznr4LuiR0HFulGTqdrbPBQ6QZSGq9x18F3hi6C6TNWVsXyPrDhppkq/vDqCihWomqk6mD+t1g8mxTtnr6qgaqYqCYL10Da7OMzuK7E6gooVqJKpSoLj4+PzO26mSu9eVUHVTNWxIojLlWhFBdVkqo6P7z1/7Gaq9gt3GuiMvgSVu2j1mGeqhAj++vV7z/4hyDNVaiNZVUGFTNXT4VX54nTt+N5P1vNMVZoNn66IoHIDSzNVl24Jm6lKgrVz5UyVXQff6SdtXVD5V4GZPldk1xbprUpGiYqwtv1plqeDGlT5VWCmAtlri2iczD49Ebe3PZSj6EqQ/VVgtgcR+bXFGwfJy5+4lugjlY90JsjcPMoIUpXFXluodiVrUPuf4u3SnSC7tEldQwT22mLbdDoUZBehjLSg4P79+0ESbwmxZ3skCnJrkPIjDe0598yiILcPMoLOu/fMoiB796zECvrXz+5vuPfMoqDCeVDw72/9/nzhnlmrLah88yjJufOF9RpRGDby2wjboe/4i/dQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQH+D5EiCIbMyJP7AAAAAElFTkSuQmCC" /><!-- --></p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
